### PR TITLE
Webchannel redux

### PR DESCRIPTION
**Pull request by jherico**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4d2087778bfafea49d219c9770ac1474aa95b3b9 available at: https://gist.github.com/e993202ab4f1503906cd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.dmg

**Comment by sethalves** [comment]
I've tried this on Linux.  Interface works, edit.js works.

**Comment by sethalves** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4d2087778bfafea49d219c9770ac1474aa95b3b9 available at: https://gist.github.com/d6919b4493ece2ff547c<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b2b2f969875923f78c3813ae4ff0a77144f1469d available at: https://gist.github.com/e993202ab4f1503906cd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0727593c4dc40d8f64a9644ce64fd56ae153084c available at: https://gist.github.com/e993202ab4f1503906cd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e058393c17e6f73e53c0ad270bf194cc61d862e6 available at: https://gist.github.com/e993202ab4f1503906cd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.dmg

File: interface/resources/qml/controls/WebView.qml [hunk 1]
```diff
@@ -59,6 +59,7 @@ WebEngineView {
             request.openIn(newWindow.webView)
     }
 
-
-    profile: desktop.browserProfile
+    // This breaks the webchannel used for passing messages.  Fixed in Qt 5.6
+    // See https://bugreports.qt.io/browse/QTBUG-49521
+    //profile: desktop.browserProfile
```
Comment (ZappoMan) [inline interface/resources/qml/controls/WebView.qml]: what's the side effect of not having this profile?

File: libraries/ui/src/QmlWindowClass.cpp [hunk 2]
```diff
@@ -20,203 +20,115 @@
...
+                object->setProperty(TITLE_PROPERTY, properties[TITLE_PROPERTY].toString());
             }
-            object->setProperty(SOURCE_PROPERTY, url);
-            if (visible) {
-                object->setProperty("visible", true);
+            if (properties.contains(HEIGHT_PROPERTY) && properties.contains(WIDTH_PROPERTY)) {
+                auto height = properties[HEIGHT_PROPERTY].toInt();
+                auto width = properties[WIDTH_PROPERTY].toInt();
+                if (width != -1 && height != -1) {
+                    width = std::max(100, std::min(1280, width));
```
Comment (ZappoMan) [inline libraries/ui/src/QmlWindowClass.cpp]: coding standard: magic numbers

File: libraries/ui/src/QmlWindowClass.cpp [hunk 3]
```diff
@@ -20,203 +20,115 @@
...
+                    height = std::max(100, std::min(720, height));
```
Comment (ZappoMan) [inline libraries/ui/src/QmlWindowClass.cpp]: magic

File: libraries/ui/src/QmlWindowClass.h [hunk 4]
```diff
@@ -13,45 +13,23 @@
...
-private:
-    const QmlWindowClass* _webWindow { nullptr };
-    QWebSocket *_socket { nullptr };
-};
 // FIXME refactor this class to be a QQuickItem derived type and eliminate the needless wrapping 
 class QmlWindowClass : public QObject {
     Q_OBJECT
-    Q_PROPERTY(QObject* eventBridge READ getEventBridge CONSTANT)
-    Q_PROPERTY(int windowId READ getWindowId CONSTANT)
+//    Q_PROPERTY(QObject* eventBridge READ getEventBridge CONSTANT)
```
Comment (ZappoMan) [inline libraries/ui/src/QmlWindowClass.h]: dead code?

**Comment by cozza13** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: e058393c17e6f73e53c0ad270bf194cc61d862e6 available at: https://gist.github.com/e993202ab4f1503906cd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.dmg

File: interface/resources/qml/controls/WebView.qml (hunk 1 above)
Comment (jherico) [inline interface/resources/qml/controls/WebView.qml]: we don't set the 'storage name' and we don't set the user agent header.  The user agent header doesn't seem to impact anything I tested (marketplace is most critical here).  The storage name may mean that it uses some default storage, or that it doesn't provide persistent storage at all, which would mean no cookies from session to session and no cache.

We explicitly set the profile in the Browser.qml so we get caching and cookies there.

File: libraries/ui/src/QmlWindowClass.cpp (hunk 2 above)
Comment (jherico) [inline libraries/ui/src/QmlWindowClass.cpp]: fixed

File: libraries/ui/src/QmlWindowClass.cpp (hunk 3 above)
Comment (jherico) [inline libraries/ui/src/QmlWindowClass.cpp]: fixed

File: libraries/ui/src/QmlWindowClass.h (hunk 4 above)
Comment (jherico) [inline libraries/ui/src/QmlWindowClass.h]: removed

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5343dfef24f364ecb55dbe2a4f04004b84b095f0 available at: https://gist.github.com/e993202ab4f1503906cd<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7480.dmg

**Comment by cozza13** [comment]
tried in windows. edit works, created web entities and they worked. teleported and moved objects. QA Pass.

**Pull request merged**